src/shared/components/ResourceDetails/ResourceDetails.js

```diff
@@ -281,15 +282,43 @@ function Resource({
     <ResourceStatusCard
       statusBadge={statusBadge ? statusBadge(resource) : null}
       customColumns={
-        <>
-          {customStatusColumns?.filter(filterColumns)?.map(col => (
-            <DynamicPageComponent.Column key={col.header} title={col.header}>
-              {col.value(resource)}
-            </DynamicPageComponent.Column>
-          ))}
-        </>
+        customStatusColumns ? (
+          <>
+            {customStatusColumns
+              ?.filter(filterColumns)
+              .filter(col => !col?.conditionComponent)
+              ?.map(col => (
+                <DynamicPageComponent.Column
+                  key={col.header}
+                  title={col.header}
+                >
+                  {col.value(resource)}
+                </DynamicPageComponent.Column>
+              ))}
+          </>
+        ) : null
       }
       conditions={statusConditions ? statusConditions(resource) : null}
+      customConditionsComponent={
+        customStatusColumns ? (
+          <>
+            {customStatusColumns
+              ?.filter(filterColumns)
+              .filter(col => col?.conditionComponent)
+              ?.map(component => (
+                <>
+                  <div
+                    className="title bsl-has-color-status-4 "
+                    style={spacing.sapUiSmallMarginBeginEnd}
+                  >
+                    {component.header}:
+                  </div>
+                  {component.value(resource)}
+                </>
+              ))}
+          </>
+        ) : null
+      }
```

Comment:
extract this filtering logic to the ExtensibilityDetails, so we don't have extensibility related logic in the component